webhook_testing_gateway.go

```diff
@@ -0,0 +1,34 @@
+package braintree
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// WebhookTestingGateway exports actions only available in the sandbox environment.
+type WebhookTestingGateway struct {
+	*Braintree
+	apiKey apiKey
+}
+
+// Request simulates an incoming webhook notification request
+func (g *WebhookTestingGateway) Request(method, endpoint, kind, id string) (*http.Request, error) {
+	payload := g.GetPayload(kind, id)
+	signature, err := g.SignPayload(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	form := url.Values{}
+	form.Add("bt_signature", signature)
+	form.Add("bt_payload", payload)
+
+	r, err := http.NewRequest(method, endpoint, strings.NewReader(form.Encode()))
```

Comment:
We should avoid use of `http.NewRequest` for building test requests that mock server requests, because the comment on that function tells us to avoid it's use. Unfortunately we can't use the comments first suggestion of `httptest` yet because that was introduced in go1.7 and this package supports go1.3, but the alternative approach is to build the struct manually and we use that pattern elsewhere.

See how we use that elsewhere here:
https://github.com/lionelbarrow/braintree-go/pull/140/files#diff-0ff71e147a689f37b1d0372bbcc591aeR18